src/node/uri_transformer.ts

```diff
@@ -0,0 +1,44 @@
+// NOTE: copied over from lib/vscode/src/vs/common/uriIpc.ts
```

Comment:
I think we should call this `uriTransformer.ts` since that seems to match up better with the other files. I know we have `coder_cloud.ts` and `proxy_agent.ts` but IMO we should rename those too.